### ISSUE TITLE
Multiple tests failing

### ISSUE THREAD

**Issue by akashsharma02** [CogRob/distributed-mapper#3]
Hi Sid, 

I tried compiling the codebase with the latest gtsam after making some changes. 
However, when I run the tests using the command specified:
```
make testDistributedMapper.run
````
I get the following errors:
```
/home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:171: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
/home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:189: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
/home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:208: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
/home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:227: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
/home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:247: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
/home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:267: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
/home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:287: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
There were 7 failures
make[3]: *** [tests/CMakeFiles/testDistributedMapper.run.dir/build.make:77: tests/CMakeFiles/testDistributedMapper.run] Error 7
make[2]: *** [CMakeFiles/Makefile2:459: tests/CMakeFiles/testDistributedMapper.run.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:466: tests/CMakeFiles/testDistributedMapper.run.dir/rule] Error 2
make: *** [Makefile:351: testDistributedMapper.run] Error 2
```

If the tests are resolved, I can submit a PR with my edits to support the latest GTSAM version as well.

**Comment by akashsharma02** [comment]
Additionally when I run the DistributedMapper on the dataset, I get the following output, is this expected:

```
./runDistributedMapper --nrRobots 4 --dataDir ../../data/example_4robots/
[optimizeRotation] Change (Robot 0): 1.79769e+308
Copying initial to optimized
```

**Comment by Xianhyy** [comment]
> Hi Sid,
> 
> I tried compiling the codebase with the latest gtsam after making some changes.
> However, when I run the tests using the command specified:
> 
> ```
> make testDistributedMapper.run
> ```
> 
> I get the following errors:
> 
> ```
> /home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:171: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
> /home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:189: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
> /home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:208: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
> /home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:227: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
> /home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:247: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
> /home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:267: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
> /home/akashsharma/Documents/projects/distributed-mapper/distributed_mapper_core/cpp/tests/testDistributedMapper.cpp:287: Failure: "Exception: Requested variable 'Z9999999' is not in this VectorValues."
> There were 7 failures
> make[3]: *** [tests/CMakeFiles/testDistributedMapper.run.dir/build.make:77: tests/CMakeFiles/testDistributedMapper.run] Error 7
> make[2]: *** [CMakeFiles/Makefile2:459: tests/CMakeFiles/testDistributedMapper.run.dir/all] Error 2
> make[1]: *** [CMakeFiles/Makefile2:466: tests/CMakeFiles/testDistributedMapper.run.dir/rule] Error 2
> make: *** [Makefile:351: testDistributedMapper.run] Error 2
> ```
> 
> If the tests are resolved, I can submit a PR with my edits to support the latest GTSAM version as well.

Have you solved it? I meet the same question.

**Comment by akashsharma02** [comment]
@Xianhyy Yes, I've made some fixes in my fork that seem to work for now: [Distributed Mapper](https://github.com/akashsharma02/distributed-mapper)
Although cannot guarantee results matching the paper.

**Comment by itzsid** [comment]
@akashsharma02, It will be great if you can create a PR. I can review the changes.

**Comment by itzsid** [comment]
Merged in #4, closing this.
